pages/api/events/index.js

```diff
@@ -0,0 +1,19 @@
+import dbConnect from "@/db/dbConnect.js";
+import Event from "@/db/models/Events";
+
+export default async function handler(request, response) {
+  await dbConnect();
+
+  switch (request.method) {
+    case "GET":
+      try {
+        const allEvents = await Event.find();
+        if (allEvents.length === 0 || !allEvents) {
+          throw new Error("Oops no events found!");
```

Comment:
```suggestion
      return response.status(404).json({ status: "Not Found" });
```
Instead of using a general error exception (Error), it would be more semantically correct to use an HTTP status code 404 (Not Found) to indicate that no events were found. The status code 404 is specifically designed for cases where a requested resource is not found, enhancing the clarity and semantics of your response.